README.md

```diff
@@ -162,39 +230,81 @@ interface IDXOptions {
 
 **Returns** `Promise<Definition>`
 
-#### .getEntryId
+#### .getEntry
 
 **Arguments**
 
 1. `definitionId: DocID`
-1. `did?: string`
+1. `did?: string = this.id`
 
-**Returns** `Promise<DocID | null>`
+**Returns** `Promise<Entry | null>`
 
-#### .getEntry
+#### .getEntryReference
 
 **Arguments**
 
 1. `definitionId: DocID`
-1. `did?: string`
+1. `did?: string = this.id`
 
 **Returns** `Promise<unknown | null>`
 
+#### .getEntryTags
+
+**Arguments**
+
+1. `definitionId: DocID`
+1. `did?: string = this.id`
+
+**Returns** `Promise<Array<string>>`
+
 #### .setEntry
 
 **Arguments**
 
+1. `definitionId: DocID`
+1. `entry: Entry`
+
+**Returns** `Promise<void>`
+
+#### .setEntryReference
+
+Sets the content of the entry reference.
+
+> The provided tags will only be set if the entry is getting created, if it already exists tags will not be changed
+
+**Arguments**
+
 1. `definitionId: DocID`
 1. `content: unknown`
+1. `tags?: Array<string> = []`
 
 **Returns** `Promise<DocID>` the `DocID` of the created content document
 
+#### .addEntryTag
+
+**Arguments**
+
+1. `definitionid: DocID`
+1. `tag: string`
+
+**Returns** `Promise<Array<string>>` the updated set of tags
+
+#### .removeEntryTag
+
+**Arguments**
+
+1. `definitionId: DocID`
+1. `tag: string`
+
+**Returns** `Promise<Array<string>>` the updated set of tags
+
 #### .addEntry
 
 **Arguments**
 
 1. `definition: Definition`
 1. `content: unknown`
+1. `tags?: Array<string> = []`
```

Comment:
Aha, you have what I mentioned above here? Or actually, what's the difference between addEntry and setEntry?

Reply:
`addEntry` creates the definition that will be used for the reference, while `setEntry` must use an existing one.
It's mostly a convenience method useful until we have constant definition IDs.

Reply:
Oh, I see. We should probably make clear in the readme that this method will likely get removed in the future 👍

Reply:
I'll remove it as it seems its usage is confusing, I don't think it's needed for common use cases anyways